src/libraries/Microsoft.PowerFx.Core/Public/CheckResult.cs

```diff
@@ -277,6 +279,13 @@ public IEnumerable<ExpressionError> Errors
             set => _errors.AddRange(value);
         }
 
+        public CheckResult SetTexlNode(TexlNode texlNode)

```

Comment:
This seems odd to allow especially publicly... What if the caller passed it a TexlNode from a different expression?

Reply:
Yeah, I realized that can happen here